### ISSUE TITLE
Support more patterns for Dreamcast UIDs

### ISSUE THREAD

**Issue by bslenul** [TheGamesDB2/Website#36]
**Is your feature request related to a problem? Please describe.**
Some DC games have serials that can't be entered on the website, I have 2 examples in mind right now:
* Gunlord: https://thegamesdb.net/game.php?id=23801 I can't enter the UID "MK1014", it won't let me save.
* Rayman 2 US: https://thegamesdb.net/game.php?id=3957 UID "17707N" can't be saved.

Screenshots of the headers to show that the serials are correct:
![image](https://github.com/user-attachments/assets/bd07ab34-1ce3-435d-bdb3-c14c5b8a492f)
![image](https://github.com/user-attachments/assets/bcf6a7cd-3b73-43fb-a8a4-fed7c4b1d445)

**Describe the solution you'd like**
My understanding of Regex is a bit limited but from my understanding this is the patterns for DC:
> T(-| |)(\\d{4,5})(D|M|ND|N)(-| |)(\\d{2}([A-Z]|)|)
MK(-| |)\\d{5}((-| |)\\d{2}([A-Z]|)|)
HDR-(\\d{4})
\\d{3}-\\d{4}(-\\d{2}|)

So I guess the 2nd one would need to be changed to `MK(-| |)\\d{4,5}((-| |)\\d{2}([A-Z]|)|)` to be able to match 4 digits.
And change the last one to `\\d{3,5}(-\\d{4}(-\\d{2}|)|[A-Z])` maybe? To match from 3 to 5 digits then either the usual pattern (`-\\d{4}(-\\d{2}|)`) or a letter (`[A-Z]`).

But again I'm not a Regex expert so I could be completely off :D

**Comment by bslenul** [comment]
Flyinghead (Flycast emulator dev) gave me a few more special cases he encountered:
> 17701D
17701N
36801N
51034
MK-0100
MK-51027A
T-40502-N

"17701D", "17701N" and "36801N" would be covered by `\\d{3,5}(-\\d{4}(-\\d{2}|)|[A-Z])` suggested in my original post.
For "51034" it would need an extra "empty" alternative after the digits, so `\\d{3,5}(-\\d{4}(-\\d{2}|)|[A-Z]|)`.
And if you're going for the new regex, then `\\d{5}([A-Z]|)` should do the trick instead of `\\d{5}[A-Z]`. Might be better since it would prevent "123" from being valid.

"MK-0100" would be covered by `MK(-| |)\\d{4,5}((-| |)\\d{2}([A-Z]|)|)` suggested in my original post.
For "MK-51027A" maybe the regex could be changed to `MK(-| |)\\d{4,5}((-| |)(\\d{2}|)([A-Z]|)|)`, by adding an "empty" alternative to the 2nd set of digits it would match.

And for "T-40502-N" change `T(-| |)(\\d{4,5})(D|M|ND|N)(-| |)(\\d{2}([A-Z]|)|)` to `T(-| |)(\\d{4,5})(-| |)(D|M|ND|N)(-| |)(\\d{2}([A-Z]|)|)`, by adding an extra "-" check after the first digits it would match.

So basically the current patterns
```
T(-| |)(\\d{4,5})(D|M|ND|N)(-| |)(\\d{2}([A-Z]|)|)
MK(-| |)\\d{5}((-| |)\\d{2}([A-Z]|)|)
HDR-(\\d{4})
\\d{3}-\\d{4}(-\\d{2}|)
```
would become either
```
T(-| |)(\\d{4,5})(-| |)(D|M|ND|N)(-| |)(\\d{2}([A-Z]|)|)
MK(-| |)\\d{4,5}((-| |)(\\d{2}|)([A-Z]|)|)
HDR-(\\d{4}) // unchanged
\\d{3,5}(-\\d{4}(-\\d{2}|)|[A-Z]|) // "123" would become valid tho, not ideal
```
or
```
T(-| |)(\\d{4,5})(-| |)(D|M|ND|N)(-| |)(\\d{2}([A-Z]|)|)
MK(-| |)\\d{4,5}((-| |)(\\d{2}|)([A-Z]|)|)
HDR-(\\d{4}) // unchanged
\\d{3}-\\d{4}(-\\d{2}|) // unchanged
\\d{5}([A-Z]|) // new one
```

**Comment by pete-oconnell** [comment]
This is now merged and implemented on the live site.
